### PR TITLE
BLUE-1996: add request states support

### PR DESCRIPTION
**Pull request by Marc-Sev**

### CODE REVIEW

File: services/requests/request_states/schemas.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+import (
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/leanspace/terraform-provider-leanspace/helper"
+)
+
+var nameRegex = regexp.MustCompile(`^[A-Z](?:[A-Z_]*[A-Z])?$`)
```
Comment (jerome-leanspace) [inline services/requests/request_states/schemas.go:11]: It seems to be the same everywhere, maybe you can factorize it and put it in a helper?

**Review by jerome-leanspace — APPROVED** [review]
I left a comment to reduce the global duplication but otherwise it looks good

**Comment by Marc-Sev** [comment]
- [x] factorized state name validation
image

File: services/requests/request_states/schemas.go (hunk 1 above)
Comment (Marc-Sev) [reply services/requests/request_states/schemas.go]: I did it, is the solution good ?

Comment (jerome-leanspace) [reply services/requests/request_states/schemas.go]: I just meant to move nameRegex in a package and make it public (by capitalizing the first name) so you could have kept `validation.StringMatch(NameRegex, "Must be a valid State name")` but you're solution also works

Comment (Marc-Sev) [reply services/requests/request_states/schemas.go]: it seemed easier to use for the user

**Pull request merged**